### PR TITLE
Blog post/Constraints or freedom? Adventure to write a perfect program

### PR DESCRIPTION
**Pull request by Aliamondo**
First draft. Didn't proofread yet, but wanted to see if this fits our blog in the first place.

### CODE REVIEW

File: _posts/2020/2020-05-15-constraints-and-freedom-writing-perfect-program.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+---
+layout: post
+title: Constraints or freedom? Adventure to write a perfect program
+date: 2020-05-15
+header_image: public/coffee-writing-computer-blogging.jpg
```
[the remark this answers is not in the document]
Comment (Aliamondo) [reply _posts/2020/2020-05-15-constraints-and-freedom-writing-perfect-program.md:5]: I did not search for an image, as the post content was more important to me. The first image you suggested fits best

**Comment by cgebken** [comment]
Review Notes:

The title and the headlines are just suggestions. 

Background for the headline changes: We try not to start a post with a headline. And the second headline just didn’t feel like a headline to me as you’ve used exactly that phrase a view lines below without making it a headline. 

Background for the authors.yml change: We’d only say “Andrey K.” if we would have two authors called Andrey.

General note: I LOVE your way to write!

File: _posts/2020/2020-05-15-constraints-and-freedom-writing-perfect-program.md [hunk 2]
```diff
@@ -38,16 +37,17 @@ public class Sum {
...
+What if the user accidentally adds a non-digit character (such as a letter) to their number?
 That would immediately raise the `InputMismatchException` and our contender for the "*Easiest program since Hello World*" title would break in a blink of an eye.
 Not something a perfect program should do.
 
-Obvious solution is to not allow any characters other than the numbers.
-We can't, however, proceed with that, as it will limit the user to the digit characters only.
-What about the real numbers with decimal points, such as the [gravity of Earth](https://en.wikipedia.org/wiki/Gravity_of_Earth){:target="_blank"} that equals to`9.8`? This number would already break the code above, by the way, as it's not an integer and Java's `Scanner` object did not expect it.
-Regardless, `9.8` is still a valid number, and we have never stated that our perfect program only works with integers, but rather with numbers in general, so we cannot let that *bring us down* (pun intended).
+The obvious solution was to not allow any characters other than numbers.
+We couldn't, however, proceed with that, as it will limit the user to the digit characters only.
```
Comment (Aliamondo) [inline _posts/2020/2020-05-15-constraints-and-freedom-writing-perfect-program.md:47]: My intention here was to have a conversation with a reader, as _both of us_ are trying to find the perfect program together

File: _posts/2020/2020-05-15-constraints-and-freedom-writing-perfect-program.md [hunk 3]
```diff
@@ -83,9 +83,8 @@ public class Sum {
 }
 ```
 
-## Well... Not really
-We run our program and start manually testing it (we don't want to write unit tests to keep our program as small as possible, but keep in mind that this is a must if we want to make our program perfect in every way).
-This gives us the following:
+We ran our program and started manually testing it (we didn't want to write unit tests to keep our program as small as possible, but keep in mind that this is a must if we want to make our program perfect in every way).
+This gave us the following:
```
Comment (Aliamondo) [inline _posts/2020/2020-05-15-constraints-and-freedom-writing-perfect-program.md:87]: Same here, trying to find the solution together with the reader to keep him entertained

File: _posts/2020/2020-05-15-constraints-and-freedom-writing-perfect-program.md [hunk 4]
```diff
@@ -113,7 +112,7 @@ Please enter the second real number
 Sum of the numbers = 3.3000000000000003
 ```
 
-The precision is actually working as intended, as the Java 8 [tutorial](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html){:target="_blank"} notes that *this data type should never be used for precise values*.
+The precision was actually working as intended, as the Java 8 [tutorial](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html){:target="_blank"} notes that *this data type should never be used for precise values*.
```
Comment (Aliamondo) [inline _posts/2020/2020-05-15-constraints-and-freedom-writing-perfect-program.md:115]: Hm. This is more like a side note, as this is the intended behavior of something. Yes, it _was_ working before, but it's supposed to _still_ work now

**Pull request merged**